src/serial.c

```diff
@@ -122,6 +131,10 @@ static void serial_in(serial_dev_t *s, uint16_t offset, void *data)
                 priv->lsr &= ~UART_LSR_DR;
                 serial_update_irq(s);
             }
+            if (fifo_level(&priv->rx_buf) == FIFO_LEN / 2)
+                /* rx_buf is half full now. Notify the worker thread to read
```

Comment:
You shall explain the reason why half full matters.

Reply:
I've added more explanation to this comment. This is the new revision taken from the updated source code:
```
/* The worker thread waits on the condition variable when rx_buf is
 * full and stdin is still readable. Notify the worker thread when
 * the level of the buffer drops to its half size, so the worker
 * thread can read up to half of the buffer size before it is
 * blocked again.
 */
```
Is this explanation OK?

Reply:
It looks better now. Refine the term "level" as I requested.